ABI File Errhandler broken

cf. https://github.com/mpi4py/mpi4py-testing/actions/runs/9039938318/job/24843498645

```python
from mpi4py import MPI

file = MPI.File.Open(MPI.COMM_SELF, "/dev/null")

def errhandler_fn(fh, ierr):
    assert fh == file, f"{file.handle=} {fh.handle=}"

eh = MPI.File.Create_errhandler(errhandler_fn)
file.Set_errhandler(eh)
file.Call_errhandler(MPI.SUCCESS)
```

```console
$ python test-eh.py
Traceback (most recent call last):
  File "src/mpi4py/MPI.src/errhimpl.pxi", line 44, in mpi4py.MPI.__pyx_fuse_3errhdl_call_mpi
    registry[index](pyhandle, errcode)
  File "/home/dalcinl/Devel/mpi4py/test-eh.py", line 6, in errhandler_fn
    assert fh == file, f"{file.handle=} {fh.handle=}"
           ^^^^^^^^^^
AssertionError: file.handle=94541399181280 fh.handle=579061728
Fatal Python error: exception in user-defined error handler function
Abort(1) on node 0 (rank 0 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
```

--- a/src/mpi/errhan/errutil.c
+++ b/src/mpi/errhan/errutil.c
@@ -189,7 +189,7 @@ int MPIR_call_errhandler(MPIR_Errhandler * errhandler, int errorcode, MPIR_handl
     } else if (h.kind == MPIR_WIN) {
         abi_handle = ABI_Win_from_mpi(h.u.handle);
     } else if (h.kind == MPIR_FILE) {
-        abi_handle = ABI_File_from_mpi(h.u.handle);
+        abi_handle = ABI_File_from_mpi(h.u.fh);
     } else if (h.kind == MPIR_SESSION) {
         abi_handle = ABI_Session_from_mpi(h.u.handle);
     } else {

